api/stats.js

```diff
@@ -0,0 +1,101 @@
+import { config } from './../config.js';
+
+export class Stats {
+    constructor(discordClient) {
+        this.client = discordClient;
+        this.stats = {};
+    }
+
+    initializeStats(prev_message) {
+        /*
+        This is a placeholder.
+        We have some limitations on how many messages we can load per request
+        we can have daily pulls, use the latest message as the last source of truth
+        and update with new stats
+        let stats = {
+            username1: {
+                messages: 5, 
+                words: 20, 
+                letters: 150
+            }
+        }
+       */
+        return {};
+    }
+
+    async _buildStats(channels, after) {
+        let stats = initializeStats();
+
+        for (let [_, channel] of channels) {
```

Comment:
yeah, this is a weird one, it's because `channel.fetch` returns a set of [id, channel] - I think that's because it is a map, we can probably do. `for(let channel of channels.values())` - will try that.